iml-gui/crate/src/page/filesystem.rs

```diff
@@ -6,33 +6,100 @@ use crate::{
     extensions::MergeAttrs,
     generated::css_classes::C,
     route::RouteId,
-    GMsg, Route,
+    sleep_with_handle, GMsg, Route,
 };
+use futures::channel::oneshot;
 use iml_wire_types::{
     warp_drive::{ArcCache, Locks},
     Filesystem, Session, Target, TargetConfParam, TargetKind, ToCompositeId,
 };
 use number_formatter as nf;
 use seed::{prelude::*, *};
+use std::time::Duration;
 use std::{collections::HashMap, sync::Arc};
 
 pub struct Row {
     dropdown: action_dropdown::Model,
 }
 
+#[derive(Default, serde::Deserialize, Clone, Debug)]
+pub(crate) struct Stats {
+    pub(crate) bytes_total: Option<u64>,
+    pub(crate) bytes_free: Option<u64>,
+    pub(crate) files_total: Option<u64>,
+    pub(crate) files_free: Option<u64>,
+    pub(crate) clients: Option<u64>,
+}
+
 pub struct Model {
     pub fs: Arc<Filesystem>,
-    pub mdts: Vec<Arc<Target<TargetConfParam>>>,
-    pub mdt_paging: paging::Model,
-    pub mgt: Vec<Arc<Target<TargetConfParam>>>,
-    pub osts: Vec<Arc<Target<TargetConfParam>>>,
-    pub ost_paging: paging::Model,
-    pub rows: HashMap<u32, Row>,
-    pub stratagem: stratagem::Model,
+    mdts: Vec<Arc<Target<TargetConfParam>>>,
+    mdt_paging: paging::Model,
+    mgt: Vec<Arc<Target<TargetConfParam>>>,
+    osts: Vec<Arc<Target<TargetConfParam>>>,
+    ost_paging: paging::Model,
+    rows: HashMap<u32, Row>,
+    stratagem: stratagem::Model,
+    stats: Stats,
+    stats_cancel: Option<oneshot::Sender<()>>,
+    stats_url: String,
+}
+
+impl Model {
+    pub(crate) fn new(fs: &Arc<Filesystem>) -> Self {
+        let query = format!(
+            r#"SELECT SUM(b_total), SUM(b_free)
+                    , SUM(f_total), SUM(f_free)
+                    , SUM(clients)
+               FROM (SELECT LAST(bytes_total) AS b_total
+                          , LAST(bytes_free) AS b_free
+                          , LAST(files_total) AS f_total
+                          , LAST(files_free) AS f_free
+                      FROM target
+                      WHERE "kind" = '"OST"' AND "fs" = '"{fs_name}"' AND time > now() - 2m
```

Comment:
`time > ...` is an attempt for https://github.com/whamcloud/integrated-manager-for-lustre/issues/1692

Reply:
Can we leave this off for now?